.github/workflows/rust.yml

```diff
@@ -60,14 +60,27 @@ jobs:
   test:
     needs: diff
     if: needs.diff.outputs.isRust == 'true'
-    runs-on: [self-hosted, self-hosted-ubuntu]
     timeout-minutes: 30
+    runs-on: ${{ matrix.os }}
     strategy:
+      matrix:
+        os:
+          - [self-hosted, self-hosted-ubuntu]
+          - [self-hosted, windows]
       fail-fast: false
     steps:
+      # Set the path for windows builds: the runner needs to be able to find
+      # llvm binaries as well as the "bash" binary included with msys64.
+      - name: Adjust path for windows builds
+        if: "startsWith(runner.os, 'windows')"
+        run: |
+          echo "C:\Program Files\llvm\bin;C:\msys64\usr\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
       - uses: actions/checkout@v3
       - uses: actions-rs/toolchain@v1
+      # The install script for nextest doesn't work on windows (very linux/bash
+      # specific) so the windows builders already have the binaries installed.
       - uses: taiki-e/install-action@nextest
+        if: "startsWith(runner.os, 'linux')"
```

Comment:
the docs for this action state that it should work on windows. is this another case of deps just not being installed on the runner?

Reply:
I tried, ends up not working well on even windows bash :shrug: